misk-feature-testing/src/main/kotlin/misk/feature/testing/FakeFeatureFlags.kt

```diff
@@ -63,6 +82,42 @@ class FakeFeatureFlags @Inject constructor(val moshi: Provider<Moshi>) : Abstrac
     return delegate.getJson(feature, key, clazz, attributes)
   }
 
+  override fun track(
```

Comment:
I would generally prefer that we don't use the tracking implementations. It generally results in less flexible solutions and unexpected complexity. I like that you expanded them too in the name of completeness, but I'd rather retire them.

Reply:
Sure thing, I'll delete the "strong" variants before merging the PR 🙂

Reply:
Done in https://github.com/cashapp/misk/pull/2293/commits/4106012b23811a6d3daae178cee6208eaef72af8 :+1: